Makefile

```diff
@@ -32,18 +32,43 @@ SUB := $(addsuffix .sub, $(ALL))
 build: build.sub
 build-linux: build-linux.sub
 build-darwin: build-darwin.sub
-clean: clean.sub
 docker: docker.sub
 lint: lint.sub
 push: push.sub
 test: test.sub
 
+clean: clean.sub
+	rm -f .dockerInit
+	rm -f .buildImage
+	docker rmi -f buildImage > /dev/null 2>&1 || true
+
+# Use this target when you want to build everything using docker containers.
+# Good for cases when you don't have the tools installed (like glide or go).
+docker-all: .buildImage
+	docker run --rm -ti \
+	  -v $(PWD):/go/src/github.com/kubernetes-incubator/service-catalog \
+	  buildimage \
```

Comment:
Can I suggest giving this a more qualified name. If I'm looking through images on my system, it's easy to forget and have no clue at all what "buildimage" is an image _for_. Can we call it something like "service-catalog-build-image" instead?